src/nvim/channel.c

```diff
@@ -517,6 +517,46 @@ uint64_t channel_from_stdio(bool rpc, CallbackReader on_output, const char **err
   return channel->id;
 }
 
+size_t newline_adjust(char **str)
```

Comment:
- given your work since this PR, do you see any way to simplify this function? e.g. we were able to avoid strktok in the other PRs.
- needs a brief docstring explaining what's going on here, with a simple concrete example like: before "...", after: "..."